src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public FigureDrawable getRandomFigure() {
+        int value = random.nextInt(4);
+        String color = colorSupplier.getRandomColor();
+
+        switch (value) {
+            case 0: {
+                return new IsoscelesTrapezoid(color, random.nextInt(10) + 1,
+                        random.nextInt(10) + 1, random.nextInt(10) + 1);
```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability.